anchors/anchor_repository_integration_test.go

```diff
@@ -3,125 +3,84 @@
-
-	confirmations, err = id.AddKeyToIdentity(context.Background(), identity.KeyPurposeEthMsgAuth, pubKey)
-	assert.Nil(t, err, "should not error out when adding keys")
-	assert.NotNil(t, confirmations, "confirmations channel should not be nil")
-	watchRegisteredIdentityKey := <-confirmations
-	assert.Nil(t, watchRegisteredIdentityKey.Error, "No error thrown by context")
-
-	return centrifugeId
-}
-
 func TestCommitAnchor_Integration(t *testing.T) {
-	anchorID, _ := hexutil.Decode("0x154cc26833dec2f4ad7ead9d65f9ec968a1aa5efbf6fe762f8f2a67d18a2d9b1")
-	documentRoot, _ := hexutil.Decode("0x65a35574f70281ae4d1f6c9f3adccd5378743f858c67a802a49a08ce185bc975")
-	centrifugeId := utils.RandomSlice(identity.CentIDLength)
-	createIdentityWithKeys(t, centrifugeId)
-	testPrivateKey, _ := hexutil.Decode("0x17e063fa17dd8274b09c14b253697d9a20afff74ace3c04fdb1b9c814ce0ada5")
+	anchorID := utils.RandomSlice(32)
+	documentRoot := utils.RandomSlice(32)
+
 	anchorIDTyped, _ := anchors.ToAnchorID(anchorID)
-	centIdTyped, _ := identity.ToCentID(centrifugeId)
 	docRootTyped, _ := anchors.ToDocumentRoot(documentRoot)
-	messageToSign := anchors.GenerateCommitHash(anchorIDTyped, centIdTyped, docRootTyped)
-	signature, _ := secp256k1.SignEthereum(messageToSign, testPrivateKey)
-	commitAnchor(t, anchorID, centrifugeId, documentRoot, signature, [][anchors.DocumentProofLength]byte{utils.RandomByte32()})
+	commitAnchor(t, anchorID, documentRoot, [][anchors.DocumentProofLength]byte{utils.RandomByte32()})
 	gotDocRoot, err := anchorRepo.GetDocumentRootOf(anchorIDTyped)
 	assert.Nil(t, err)
 	assert.Equal(t, docRootTyped, gotDocRoot)
 }
 
-func commitAnchor(t *testing.T, anchorID, centrifugeId, documentRoot, signature []byte, documentProofs [][32]byte) {
+func commitAnchor(t *testing.T, anchorID, documentRoot []byte, documentProofs [][32]byte) {
 	anchorIDTyped, _ := anchors.ToAnchorID(anchorID)
 	docRootTyped, _ := anchors.ToDocumentRoot(documentRoot)
-	centIdFixed, _ := identity.ToCentID(centrifugeId)
 
-	cfg.Set("identityId", centIdFixed.String())
 	ctx := testingconfig.CreateAccountContext(t, cfg)
-	confirmations, err := anchorRepo.CommitAnchor(ctx, anchorIDTyped, docRootTyped, centIdFixed, documentProofs, signature)
-	if err != nil {
-		t.Fatalf("Error commit Anchor %v", err)
-	}
+	done, err := anchorRepo.CommitAnchor(ctx, anchorIDTyped, docRootTyped, documentProofs)
+
+	isDone := <- done
+
+	assert.True(t, isDone, "isDone should be true")
+
+	assert.Nil(t, err)
 
-	watchCommittedAnchor := <-confirmations
-	assert.Nil(t, watchCommittedAnchor.Error, "No error should be thrown by context")
-	assert.Equal(t, watchCommittedAnchor.CommitData.AnchorID, anchorIDTyped, "Resulting anchor should have the same ID as the input")
-	assert.Equal(t, watchCommittedAnchor.CommitData.DocumentRoot, docRootTyped, "Resulting anchor should have the same document hash as the input")
 }
 
+/*
+
+//TODO can be removed because CommitAnchor is sync
```

Comment:
even though we it might not be relevant for our use case of anchoring, I would like to keep it here, as it tests out our tx pool mechanism. Unless we cannot trigger it anymore?